sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/expressions/aggregate/datasketchesAggregates.scala

```diff
@@ -286,19 +285,17 @@ case class HllUnionAgg(
   }
 
   /**
-   * Helper method to compare lgConfigKs and throw an exception if
-   * allowDifferentLgConfigK isn't true and configs don't match
+   * Helper method to compare lgConfigKs and throw an exception if allowDifferentLgConfigK isn't
+   * true and configs don't match.
    *
    * @param left An lgConfigK value
    * @param right An lgConfigK value
    */
   def compareLgConfigK(left: Int, right: Int): Unit = {
     if (!allowDifferentLgConfigK) {
       if (left != right) {
-        throw new UnsupportedOperationException(
-          s"Sketches have different lgConfigK values: $left and $right. " +
-            "Set allowDifferentLgConfigK to true to enable unions of " +
-            "different lgConfigK values.")
+        throw QueryExecutionErrors.hllUnionDifferentLogK(
+          left.toString, right.toString, function = "HLL_UNION_AGG")
```

Comment:
Could you just pass `left` and `right` as is, and quote them by `toSQLVal`

Reply:
Sure, done.